server/coordinator/eventdispatch/dispatch_impl.go

```diff
@@ -0,0 +1,125 @@
+// Copyright 2022 CeresDB Project Authors. Licensed under Apache-2.0.
+
+package eventdispatch
+
+import (
+	"context"
+	"sync"
+
+	"github.com/CeresDB/ceresdbproto/pkg/metaeventpb"
+	"github.com/CeresDB/ceresmeta/server/cluster"
+	"github.com/CeresDB/ceresmeta/server/service"
+	"github.com/pkg/errors"
+	"google.golang.org/grpc"
+)
+
+type DispatchImpl struct {
+	conns sync.Map
+}
+
+func NewDispatchImpl() *DispatchImpl {
+	return &DispatchImpl{}
+}
+
+func (e *DispatchImpl) OpenShard(ctx context.Context, addr string, request *OpenShardRequest) error {
+	client, err := e.getMetaEventClient(ctx, addr)
+	if err != nil {
+		return err
+	}
+	resp, err := client.OpenShard(ctx, &metaeventpb.OpenShardRequest{
+		Shard: cluster.ConvertShardsInfoToPB(request.Shard),
+	})
+	if err != nil {
+		return errors.WithMessage(err, "open shard")
+	}
+	if resp.GetHeader().Code != 0 {
+		return errors.Errorf("failed to open shard, err:%s", resp.GetHeader().GetError())
```

Comment:
The leaf error should follow the [style](https://github.com/CeresDB/ceresmeta/blob/main/docs/style_guide.md#error-handling)